subprojects/gstreamer/libs/gst/check/gstharness.c

```diff
@@ -236,6 +237,35 @@ gst_harness_chain (GstPad * pad, GstObject * parent, GstBuffer * buffer)
   return GST_FLOW_OK;
 }
 
+static GstFlowReturn
```

Comment:
The pull and chain functions are 90% copy paste of the corresponding buffer (non list) functions.